Make drizzle_binlog_get_filename return end position

andreas-bok-sociomantic opened the pull request:
Modify the function so the end position of the file is returned

andreas-bok-sociomantic committed:
Modify drizzle_binlog_get_filename function

Add func parameter to drizzle_binlog_get_filename
to save the size of the binlog file into

Make the function return a drizzle_return_t value
for better client handling of errors

Modify unittest to new function signature

mathias-lang-sociomantic commented:
That's definitely not a point release. Actually it should even be a [major](https://community.kde.org/Policies/Binary_Compatibility_Examples#Change_the_return_type).

andreas-bok-sociomantic commented:
> That's definitely not a point release. Actually it should even be a major.

Add to the API yes, but doesn't break anything.

andreas-bok-sociomantic commented:
Would concede that it might should go into to a minor release.

andreas-bok-sociomantic committed:
Update rst documentation for drizzle_binlog_get_filename

andreas-bok-sociomantic commented:
- Changed the function signature so it returns a drizzle_return_t` value
- Added unittest of the changed function signature 
- Updated RST documentation 
- Rebased on latest v6.x.x

andreas-bok-sociomantic commented:
#103 should be merged first into v6.x.x

mathias-lang-sociomantic commented:
> #103 should be merged first into v6.x.x

Still targeting v5.x.x though ?...

andreas-bok-sociomantic commented:
> Still targeting v5.x.x though ?...

Fixed

andreas-bok-sociomantic commented:
More comments?

ben-palmer-sociomantic commented:
> More comments?

It is still blocked at the moment, so is it ready for review?

andreas-bok-sociomantic commented:
> It is still blocked at the moment, so is it ready for review?

Yes

ben-palmer-sociomantic commented on libdrizzle-5.1/binlog.h:
```diff
@@ -203,23 +203,32 @@ DRIZZLE_API
- * @param[in,out] filename buffer to copy filename to
- * @param[in] file_index index of the binlog to retrieve.
+ * @param[in] con Drizzle structure previously initialized with
+ * drizzle_create() or drizzle_clone().
```
No more `drizzle_clone` right?

ben-palmer-sociomantic commented on libdrizzle-5.1/binlog.h:
```diff
@@ -203,23 +203,32 @@ DRIZZLE_API
+ * drizzle_create() or drizzle_clone().
+ * @param[in,out] filename Buffer to copy filename to
+ * @param[in,out] end_position Variable to save the size of the binlog file into
+ * @param[in] file_index: Index of the binlog to retrieve.
```
Most of the parameters don't have `:` in them.

ben-palmer-sociomantic reviewed (commented):
I had a couple of comments on the documentation otherwise LGTM.

andreas-bok-sociomantic commented:
Addressed comments by @ben-palmer-sociomantic 

Before merging: 
We might consider another change so the client is required to provide a preallocated buffer to copy   the filename into. Would make mem leaks less likely.
@mathias-lang-sociomantic

ben-palmer-sociomantic commented:
> We might consider another change so the client is required to provide a preallocated buffer to copy the filename into. Would make mem leaks less likely.

Whats the way other methods in the library do it?

andreas-bok-sociomantic commented:
> Whats the way other methods in the library do it?

[`drizzle_escape_str`](https://github.com/sociomantic-tsunami/libdrizzle-redux/blob/v5.6.x/libdrizzle/query.cc#L60) doesn't take a preallocated buffer. 
That's the only other API method with similar behavior.

ben-palmer-sociomantic commented:
> drizzle_escape_str doesn't take a preallocated buffer. That's the only other API method with similar behavior.

In that case I am fine with keeping it as it is.

ben-palmer-sociomantic commented on libdrizzle/binlog.cc:
```diff
@@ -449,44 +449,52 @@ const char *drizzle_binlog_event_type_str(drizzle_binlog_event_types_t event_typ
   }
 
   drizzle_result_free(result);
+  return DRIZZLE_RETURN_OK;
```
Is it possible that with `drizzle_result_free(result);` only being called here we could end up returning an error code without freeing the result?

andreas-bok-sociomantic replied:
Indeed, that would be the case. Will fix

ben-palmer-sociomantic commented:
Just one more comment then LGTM.

andreas-bok-sociomantic commented:
Addressed comment from @ben-palmer-sociomantic about possible memory leak. 

```diff
diff --git a/libdrizzle/binlog.cc b/libdrizzle/binlog.cc
index 9d30a5a..8a27b92 100644
--- a/libdrizzle/binlog.cc
+++ b/libdrizzle/binlog.cc
@@ -464,6 +464,7 @@ drizzle_return_t drizzle_binlog_get_filename(drizzle_st *con, char **filename,
   if (driz_ret != DRIZZLE_RETURN_OK)
   {
     drizzle_set_error(con, __FILE_LINE_FUNC__, "Query to retrieve binary logs failed");
+    drizzle_result_free(result);
     return driz_ret;
   }
 
@@ -473,6 +474,7 @@ drizzle_return_t drizzle_binlog_get_filename(drizzle_st *con, char **filename,
   if (file_index < -1 || file_index >= row_count)
   {
     drizzle_set_error(con, __FILE_LINE_FUNC__, "Invalid binlog file index: %d", file_index);
+    drizzle_result_free(result);
     return DRIZZLE_RETURN_INVALID_ARGUMENT;
   }
```
